airbyte-cdk/java/airbyte-cdk/db-sources/src/main/java/io/airbyte/cdk/integrations/debezium/internals/DebeziumStateDecoratingIterator.java

```diff
@@ -140,7 +141,7 @@ protected AirbyteMessage computeNext() {
 
     if (cdcStateHandler.isCdcCheckpointEnabled() && sendCheckpointMessage) {
       LOGGER.info("Sending CDC checkpoint state message.");
-      final AirbyteMessage stateMessage = createStateMessage(checkpointOffsetToSend);
+      final AirbyteMessage stateMessage = createStateMessage(checkpointOffsetToSend, recordsLastSync);
```

Comment:
we just need to verify that this works for GLOBAL state as well. I've cc'd you in a comment here : 
https://docs.google.com/document/d/1bO__n57LVRMba1TDoZrdaDNbUrgqKnyicaT8DiQxiGI/edit